pyanaconda/image.py

```diff
@@ -168,7 +169,8 @@ def opticalInstallMedia(devicetree):
 # Return a list of Device instances that may have HDISO install media
 # somewhere.  Candidate devices are simply any that we can mount.
 def potentialHdisoSources(devicetree):
-    return filter(lambda d: d.format.exists and d.format.mountable, devicetree.getDevicesByType("partition"))
+    return filter(lambda d: d.format.exists and d.format.mountable,
+                  (d for d in devicetree.devices if d.type == "partition"))
```

Comment:
The ``filter`` call and the ``lambda`` can be dropped here with all moved to the generator conditions.